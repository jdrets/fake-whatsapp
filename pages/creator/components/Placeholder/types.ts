export interface PlaceholderTypes {
  onSubmit: (text: string, owner: string) => void
  to: string
}
