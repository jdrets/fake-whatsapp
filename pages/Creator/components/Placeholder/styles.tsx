import styled from 'styled-components'

export const Input = styled.input`
  border-radius: 24px;
  border: 2px dotted ${({ theme }) => theme.colors.primary};
  padding: 8px 0;
  font-size: 14px;
  background: transparent;
  outline: none;
  text-align: center;
  word-break: break-word;
  
  &::placeholder {
    text-align: center;
  }
  
  &:focus {
    border-style: solid;
  }
`
