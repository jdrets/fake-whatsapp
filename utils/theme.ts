export const theme = {
  colors: {
    primary: '#3DDC97'
  }
}