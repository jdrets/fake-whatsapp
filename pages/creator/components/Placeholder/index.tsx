import React, { useState, FunctionComponent, ChangeEvent } from 'react'

import { Input } from './styles'
import { PlaceholderTypes } from './types'

const Placeholder: FunctionComponent<PlaceholderTypes> = ({ onSubmit, to }) => {
  const [text, setText] = useState('')

  const handleChangeText = (event: ChangeEvent<HTMLInputElement>) => {
    const { target: { value } } = event
    setText(value)
  }

  const handleSave = () => {
    if (text) {
      onSubmit(text, to)
      setText('')
    }
  }

  return (
    <Input
      type="text"
      value={text || ''}
      onChange={handleChangeText}
      onBlur={handleSave}
      placeholder="Agregar texto"
    />
  )
}


export default Placeholder
