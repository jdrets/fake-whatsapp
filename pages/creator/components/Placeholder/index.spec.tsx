import React from 'react'
import {render, fireEvent, waitFor} from '@testing-library/react'
import { ThemeProvider } from 'styled-components';

import { theme } from 'utils/theme'
import Placeholder from '.'

const setup = (mock = {}) => (
  render(
    <ThemeProvider theme={theme}>
      <Placeholder
        onSubmit={jest.fn()}
        to="RECEIVER"
        {...mock}
      />
    </ThemeProvider>
  )
)

describe('<Placeholder />', () => {
  test('Should match with a snapshot when owner is RECEIVER', () => {
    const { container } = setup({
      owner: 'RECEIVER'
    })

    expect(container).toMatchSnapshot()
  })

  test('Should change text when input is filled', async () => {
    const { getByPlaceholderText } = setup()
    const input = getByPlaceholderText('Agregar texto') as HTMLInputElement

    fireEvent.change(input, {
      target: {
        value: 'some dummy value'
      }
    })

    expect(input.value).toBe('some dummy value')
  })

  test('Should call onSubmit function when blur input', async () => {
    const onSubmitMock = jest.fn()
    const { getByPlaceholderText } = setup({
      onSubmit: onSubmitMock
    })
    const input = getByPlaceholderText('Agregar texto')

    fireEvent.change(input, {
      target: {
        value: 'some dummy value'
      }
    })

    fireEvent.blur(input)

    expect(onSubmitMock).toHaveBeenCalled()
  })

  test('Should call onSubmit function when enter button is pressed', async () => {
    const onSubmitMock = jest.fn()
    const { getByPlaceholderText } = setup({
      onSubmit: onSubmitMock
    })
    const input = getByPlaceholderText('Agregar texto')

    fireEvent.change(input, {
      target: {
        value: 'some dummy value'
      }
    })

    fireEvent.keyDown(input, {
      key: 'Enter'
    })

    expect(onSubmitMock).toHaveBeenCalled()
  })

  test('Should not call onSubmit function when text state is null', async () => {
    const onSubmitMock = jest.fn()
    const { getByPlaceholderText } = setup({
      onSubmit: onSubmitMock
    })
    const input = getByPlaceholderText('Agregar texto')

    fireEvent.blur(input)

    expect(onSubmitMock).toHaveBeenCalledTimes(0)
  })
})
