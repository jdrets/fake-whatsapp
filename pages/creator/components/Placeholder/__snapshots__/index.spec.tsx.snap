// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Placeholder /> Should match with a snapshot when owner is RECEIVER 1`] = `
<div>
  <input
    class="styles__Input-sc-1cvmjof-0 ncbnf"
    placeholder="Agregar texto"
    type="text"
    value=""
  />
</div>
`;
